src/components/hv-picker-field/index.js

```diff
@@ -0,0 +1,247 @@
+
+    return React.createElement(Picker, props, ...children);
+  };
+
+  /**
+   * Renders a bottom sheet with cancel/save buttons and a picker component.
+   * Uses styles defined on the <picker-field> element for the modal and buttons.
+   */
+  renderPickerModal = () => {
+    const { element, stylesheets, options } = this.props;
+    const modalStyle = createStyleProp(element, stylesheets, {
+      ...options,
+      styleAttr: 'modal-style',
+    });
+    const cancelTextStyle = createStyleProp(element, stylesheets, {
+      ...options,
+      pressed: this.state.cancelPressed,
+      styleAttr: 'modal-text-style',
+    });
+    const saveTextStyle = createStyleProp(element, stylesheets, {
+      ...options,
+      pressed: this.state.savePressed,
+      styleAttr: 'modal-text-style',
+    });
+    const cancelLabel = element.getAttribute('cancel-label') || 'Cancel';
+    const saveLabel = element.getAttribute('save-label') || 'Save';
+
+    return (
+      <Modal
+        animationType="slide"
+        transparent
+        visible={this.state.showPicker}
+        onRequestClose={this.cancelPickerModal}
+      >
+        <View
+          style={{
+            flex: 1,
+            justifyContent: 'flex-end',
+          }}
+        >
+          <View style={modalStyle}>
+            <View
+              style={{
+                flexDirection: 'row',
+                justifyContent: 'space-between',
+              }}
+            >
+              <TouchableWithoutFeedback
+                onPressIn={() => this.setState({ cancelPressed: true })}
+                onPressOut={() => this.setState({ cancelPressed: false })}
+                onPress={this.cancelPickerModal}
+              >
+                <Text style={cancelTextStyle}>{cancelLabel}</Text>
+              </TouchableWithoutFeedback>
+              <TouchableWithoutFeedback
+                onPressIn={() => this.setState({ savePressed: true })}
+                onPressOut={() => this.setState({ savePressed: false })}
+                onPress={this.savePickerModal}
+              >
+                <Text style={saveTextStyle}>{saveLabel}</Text>
+              </TouchableWithoutFeedback>
+            </View>
+            {this.renderPicker()}
+          </View>
+        </View>
+      </Modal>
+    );
+  };
+
+  render() {
+    const { element, stylesheets, options } = this.props;
+    if (element.getAttribute('hide') === 'true') {
+      return null;
+    }
+
+    const focused = this.state.showPicker;
+    const pressed = this.state.fieldPressed;
+    const props = createProps(element, stylesheets, {
+      ...options,
+      focused,
+      pressed,
+      styleAttr: 'field-style',
+    });
+    const fieldTextStyle = createStyleProp(element, stylesheets, {
+      ...options,
+      focused,
+      pressed,
+      styleAttr: 'field-text-style',
+    });
+    const value = element.getAttribute('value');
+    const placeholderTextColor = element.getAttribute('placeholderTextColor');
```

Comment:
we typically have used dash case for attributes
```suggestion
    const placeholderTextColor = element.getAttribute('placeholder-text-color');
```

Reply:
I thought about that, but we already use `placeholderTextColor` camel-cased for the `<text-field>` component. I thought it would be more consistent to keep the same attribute.

long term, I think we should support an `empty` modifier in style sheets rather than specifying the placeholder style directly as an attribute.